### PR TITLE
Expose Repo.get_ref

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by samoht** [comment]
The `refs` function is expecting an array of refs, whereas `get_ref` just expect one ref.

File: lib/github_s.mli [hunk 1]
```diff
@@ -737,6 +737,15 @@ module type Github = sig
         {{:https://developer.github.com/v3/git/refs/}git references}
         with prefix [?ty] for repo [user]/[repo]. *)
 
+    val get_ref :
+      ?token:Token.t ->
+      user:string -> repo:string ->
+      name:string ->
+      unit -> Github_t.git_ref Response.t Monad.t
+    (** [get_ref ~user ~repo ~name] is the Git reference
+        {{:https://developer.github.com/v3/git/refs/}git references}
```
Comment (dsheets) [inline lib/github_s.mli]: This doc markup repeats "Git reference git references".

File: lib/github_s.mli [hunk 2]
```diff
@@ -737,6 +737,15 @@ module type Github = sig
         {{:https://developer.github.com/v3/git/refs/}git references}
         with prefix [?ty] for repo [user]/[repo]. *)
 
+    val get_ref :
+      ?token:Token.t ->
+      user:string -> repo:string ->
+      name:string ->
+      unit -> Github_t.git_ref Response.t Monad.t
```
Comment (dsheets) [inline lib/github_s.mli]: What if the ref is not found?

Comment (samoht) [inline lib/github_s.mli]: you can an error like for `get_tag`

**Comment by samoht** [comment]
I've fixed the doc. Not sure about the other comment, I have just copied what we do for `get_tag`.

**Comment by samoht** [comment]
(e.g. it's the responsibility of the caller to catch any exception in that case)

File: lib/github_s.mli (hunk 2 above)
Comment (dsheets) [inline lib/github_s.mli]: I'd really like to get rid of those 404 errors everywhere, eventually. I guess this is OK for now but I'd appreciate an issue that starts to track all of the horrible 404 exceptions for 3.0.

**Pull request merged**